integration-test/shadow/shadow_system_test.c

```diff
@@ -659,6 +668,144 @@ static MQTTStatus_t publishToTopic( MQTTContext_t * pContext,
+
+    /* Subscribe to shadow topic /get/rejected with Qos 0. */
+    topicLength = createTopicString( ShadowTopicStringTypeGetRejected, pShadowName, shadowNameLength );
+    TEST_ASSERT_EQUAL( MQTTSuccess, subscribeToTopic( &context,
+                                                      topicString,
+                                                      topicLength,
+                                                      MQTTQoS0 ) );
+
+    /* Subscribe to shadow topic /update/accepted with Qos 0. */
+    topicLength = createTopicString( ShadowTopicStringTypeUpdateAccepted, pShadowName, shadowNameLength );
+    TEST_ASSERT_EQUAL( MQTTSuccess, subscribeToTopic( &context,
+                                                      topicString,
+                                                      topicLength,
+                                                      MQTTQoS0 ) );
+
+    /* Subscribe to shadow topic /update/rejected with Qos 0. */
+    topicLength = createTopicString( ShadowTopicStringTypeUpdateRejected, pShadowName, shadowNameLength );
+    TEST_ASSERT_EQUAL( MQTTSuccess, subscribeToTopic( &context,
+                                                      topicString,
+                                                      topicLength,
+                                                      MQTTQoS0 ) );
+
+    /* Subscribe to shadow topic /update/delta with Qos 0. */
+    topicLength = createTopicString( ShadowTopicStringTypeUpdateDelta, pShadowName, shadowNameLength );
+    TEST_ASSERT_EQUAL( MQTTSuccess, subscribeToTopic( &context,
+                                                      topicString,
+                                                      topicLength,
+                                                      MQTTQoS0 ) );
+
+    /* Subscribe to shadow topic /update/documents with Qos 0. */
+    topicLength = createTopicString( ShadowTopicStringTypeUpdateDocuments, pShadowName, shadowNameLength );
+    TEST_ASSERT_EQUAL( MQTTSuccess, subscribeToTopic( &context,
+                                                      topicString,
+                                                      topicLength,
+                                                      MQTTQoS0 ) );
+
+    /* First of all, try to delete any Shadow document in the cloud.
+     * This could trigger the /delete/accepted or /delete/rejected
+     * based on the thing status on the cloud.
+     */
+    ( void ) createTopicString( ShadowTopicStringTypeDelete, pShadowName, shadowNameLength );
+    TEST_ASSERT_EQUAL( MQTTSuccess, publishToTopic( &context,
+                                                    topicString,
+                                                    updateDocument,
+                                                    MQTTQoS0 ) );
+
+    /* Check the flag for /delete/accepted or /delete/rejected. */
+    TEST_ASSERT_TRUE( ( receivedDeleteAcceptedResult || receivedDeleteRejectedResult ) );
```

Comment:
It will be better to rest both these flags before publish so as to confirm if they are toggled. Applies to the similar asserts below also.

Reply:
The test sequence will receive just one of delete/accepted or delete/rejected, depdnding on whether the shadow exists or not. (ie. First run of the sequence for this thing/sahdow or not.) So I think the OR here is correct.

For the UPDATE and GET operations, I don't see that a rejected response should ever be received, if the test sequence is successful. So I have deleted those two ORs.

Reply:
My bad. It is a typo that caused the confusion. "rest" in the comment misses an "e". It is reset. So the comment is 

"It will be better to res**e**t both these flags before publish so as to confirm if they are toggled. Applies to the similar asserts below also."

Please let me know if you have any further questions about it.

Reply:
I realized it was a typo, but read it as "test", not "reset". 

All flags are reset in the setup(), so I think it should not be necessary to reset them inline in the test case. If anything, I would suggest to just add asserts that they are false, before doing the publish.